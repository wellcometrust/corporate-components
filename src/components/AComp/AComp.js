import React from 'react';

export const AComp = ({ children }) => (
  <div className="acomp">
    <h2 className="acomp__title">Test Component</h2>
    <div className="acomp__content">{children}</div>
  </div>
);

export default AComp;
