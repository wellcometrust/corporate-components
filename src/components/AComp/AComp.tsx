import React from 'react';

type ACompProps = {
  children?: any
}

export const AComp = ({children}: ACompProps) =>(
<div className="acomp"><h2 className="acomp__title">Test Component</h2><div className="acomp__content">{children}</div></div>
)

export default AComp;
