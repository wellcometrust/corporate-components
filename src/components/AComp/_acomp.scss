.acomp {
  align-items: center;
  background: var(--colour-grey-80);
  color: var(--colour-grey-05);
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin: var(--space-md);
  padding: var(--space-xl);
  z-index: 1;
}

  .acomp__title {
    color: white;
  }

  .acomp__content {
    border: 1px solid black;

    &::before {
      content: "";
    }
  }
