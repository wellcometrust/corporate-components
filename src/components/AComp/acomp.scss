@import '../../styles/colours';

.acomp {
  align-items: center;
  background: $blue60;
  color: #555;
  display: flex;
  flex-flow: row wrap;
  justify-content: space-between;
  z-index: 1;
}

.acomp__title {
  color: black;
}

.acomp__content {
  border: 1px solid #66f;

  &::before {
    content: "";
  }
}
