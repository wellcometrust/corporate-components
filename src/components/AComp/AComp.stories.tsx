import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean, text, select } from '@storybook/addon-knobs';

import AComp from './AComp';

const ACompExample = ({ children }) => {
  const compText = text('compText', 'Hello, world!');

  return <AComp>{compText}</AComp>;
};

const stories = storiesOf('Components|AComp', module);

stories.add('AComp', ACompExample);
