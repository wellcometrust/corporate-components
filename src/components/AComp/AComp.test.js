import React from 'react';
import { shallow } from 'enzyme';
import AComp from './AComp';

describe('<AComp />', () => {
  const output = shallow(<AComp>Button text</AComp>);

  it('renders the component', () => {
    expect(output);
  });
});
