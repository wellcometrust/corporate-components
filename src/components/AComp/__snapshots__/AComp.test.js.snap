// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AComp /> renders the component 1`] = `ShallowWrapper {}`;
